### ISSUE TITLE
Working with Magento 1.9.1.0?

### ISSUE THREAD

**Issue by winkelsdorf** [yireo/Yireo_AdminPreviousNext#3]
After the composer.json configuration I totally forgot to test this Extension. Does not seem to work anymore with 1.9.x, using a fresh install.

Can you confirm this?

**Comment by yireo** [comment]
I've just checked this in a fresh Magento 1.9.1.0 site myself, and it works like a charm. Are you seeing the buttons? Which extension source did you install from? Are the files there on the file level?

**Comment by winkelsdorf** [comment]
Installed it via Composer (Satis pointing to here). 

I found the mistake: In the modman you are not copying the skin\ directory at all.

I am preparing a patch right now, additionally I would like to see the skin/adminhtml/default files to be under a subdirectory (like /skin/adminhtml/default/default/adminhtml). Think this might be best practice?

Creating a pull request for you shortly after both changes are completed and tested :)

**Comment by winkelsdorf** [comment]
Ok, fixed modman and adjusted the skin paths.

One question: Why do you trigger load of jQuery via backend setting? It's required for this extension, so probably best to load it.. I am creating a patch which respects jQuery.noConflict() right now..

**Comment by winkelsdorf** [comment]
Fixed another bug in Order.php. See my current pull request.

**Comment by yireo** [comment]
Hi, thanks for all the work.

As for the path `skin/adminhtml/default/default/adminhtml` - I'm not sure what that would, the name "adminhtml" would be duplicated. As to my experience, there are two ways extensions can place their own files in that folder: By using their own base-folder `skin/adminhtml/default/default/EXTENSION` and then place subfolders `css` or `js` inside. Or by using a base-folder `skin/adminhtml/default/default/css/EXTENSION` (here only for CSS). I've never seen a strategy like yours before ...

As for jQuery: The most common mistake of loading jQuery in either the frontend or backend, is not to have backend setting. If there is more than 1 extension that loads jQuery, you will have more than 1 jQuery library in your HTML, which first of all does not make sense, second it creates a performance issue, third it allows for further JS conflicts. Why do you consider this is an issue? I'm sure you must have seen this with other extensions as well, or that you have struggled with duplicate jQuery loads before?

As for the issue in Order.php, this is stupid mistake I recall having fixed ages ago. Strange that it is still there.

Because of the numerous issues, that are very apart from each other, I need to decline the PR and manually copy changes instead. The bad thing of having 1 PR with multiple fixes that don't have a common ground, is that you can't accept a single fix. Therefore it is always best to have various commits (which you did) but various PRs as well.

I've done the copying now - let me know what you think.

**Comment by winkelsdorf** [comment]
Hi,

Regarding the PR: I agree, that was a little mess with too many changes at once. I'll create single change pull requests in future.

Your changes look good except for one bug:

**Bug Regarding the paths**
There's an error with the modman file with your latest commit: 
1) Either you adjust the css/js/image files to be stored under /source/skin/adminhtml/default/default/adminpreviousnext in your Repo. That's what the current modman needs.
or
2) You need to edit the modman again to fit your own structure: /source/skin/adminhtml/default/default/*.

You are absolutely right about the two possibilites for storing the files. For the adminhtml/.../adminhtml thing: Sorry, that was my fault in an early commit and PR. I fixed it in my last commit, you should have received the pull request for https://github.com/winkelsdorf/Yireo_AdminPreviousNext/commit/2c59e219f7f88ed3a4b13d4976075a4ff9166d4d which includes all fixes.

Anyway, I would suggest to use the 1st variant, as you said: `By using their own base-folder skin/adminhtml/default/default/EXTENSION`.

Why? Because it's simply safer. E.g. you have a file called "next.css" and store it directly under adminhtml/default/default/css/next.css. Now another extension wants to do the same, because it also has a backend job with a previous/next situation. That developer might also pick next.css as name and wants to put it there too. Happy debugging ;) 

_So I think better follow your 1st suggestion and change the paths in the Repo._

``` code
Loading composer repositories with package information
Updating dependencies (including require-dev)                                
  - Installing yireo/yireo_adminpreviousnext (dev-master 8960233)
    Cloning 8960233908595befe6fc5b75a69901c32667205a

Writing lock file
Generating autoload files

  [ErrorException]                                                                                                     
  Source /Users/winkelsdorf/Sites/magentotest.dev/vendor/yireo/yireo_adminpreviousnext/source/skin/adminhtml/default/default/adminpreviousnext does not exist
```

That's because of the difference between modman and path structure of course.

tl;dr
- Modman is correct now
- Skin Folders are not
- Other fixes look good

Again, many thanks for your work!

Cheers,
Frederik

**Comment by winkelsdorf** [comment]
Addendum (if you don't read here but via mail):

Watch out if you change the path, then you need to edit 
- **adminpreviousnext.xml** (https://github.com/yireo/Yireo_AdminPreviousNext/blob/master/source/app/design/adminhtml/default/default/layout/adminpreviousnext.xml). There's the **addItem reference** for adding the css/js to the head.
- **next.css** (https://github.com/yireo/Yireo_AdminPreviousNext/blob/master/source/skin/adminhtml/default/default/css/adminpreviousnext/next.css). Change the **relative reference** from '../../' to '../'.

**jQuery**
Edit: Forgot to answer regarding jQuery. I tried to include a .noConflict call to jQuery but your answer makes sense and it's probably the easiest way to solve that problem.

But do you need jQuery at all? You are doing a very elegant small DOM manipulation in previous/next.phtml. If you are speaking about performance/conflicts: Pick another library, way smaller. Have a look at the collection on http://microjs.com/#dom and what I would suggest, pick any small library of your choice.

I did a test commit on my forked repo now which includes DOMtastic (http://webpro.github.io/DOMtastic/) together with a .noConflict resolution and it works fine :)

The generated code is beautiful as:

``` code
<script>
    var yireoDomtastic = $.noConflict();
</script>

<script>
var previousHtml = '<button type="button" class="back" '
    + 'title="test4" '
    + 'onclick="setLocation(\'https://magentotest.dev/index.php/myadmin/catalog_product/edit/id/4/key/138c1bf5fbbe46775cbde10e3f747b01/\');">'
    + '<span><span><span>Vorheriger</span></span></span>'
    + '</button>';
yireoDomtastic('p.form-buttons').append(previousHtml);
</script>
```

This library is a) highly compatibly (even old browser while current jQuery started skipping them) b) 14 kb, leightweight and fast and c) the .noConflict makes it compatibly with every jQuery/Scriptalicious/you name it library on Magento Backend/Frontend.

Going to do a 2nd commit removing the backend options.

I promise, I won't create a PR unless you tell me to ;) Have a look at my fork: https://github.com/winkelsdorf/Yireo_AdminPreviousNext/commits/develop

Edit 2: My changes broke something. I will publish a fix asap.
Edit 3: Everything working now. Needed to call.noConflict directly after inclusion of Domtastic. Fixed :)

See my today's commits: https://github.com/winkelsdorf/Yireo_AdminPreviousNext/commits/develop

**Comment by jissereitsma** [comment]
Hi,

Sorry for the long delay. I think the CSS and JS locations as they are right now (skin/adminhtml/default/default/css/adminpreviousnext and skin/adminhtml/default/default/js/adminpreviousnext) don't create any conflict. As for the jQuery comment: Indeed, it is a bit strange that jQuery is required for such a small thing. While it is a matter of personal taste (I really don't like ProtoType), writing an effect this small in the right framework is trivial. So instead of introducing any third party framework (jQuery, Domtastic) I would say the effort should be put into getting this working with ProtoType. I've added that to my todo-list.

**Comment by winkelsdorf** [comment]
Agreed, it's looking good as it is now. I wouldn't say it needs a rewrite using ProtoType anymore (I don't like ProtoType, too) due to the fact that Magento 2 will remove ProtoType completely and support jQuery by default.

tl;dr
No need for this to be on your todo-list ;)
